integration_tests/features/Sync.feature

```diff
@@ -100,3 +100,20 @@ Feature: Block Sync
     And I mine 6 blocks on PNODE2
     When I start NODE1
     Then all nodes are at height 20
+
+  @long-running
+  Scenario: Syncing node while also mining before tip sync
+    Given I have a seed node SEED
+    And I have wallet WALLET1 connected to seed node SEED
+    And I have wallet WALLET2 connected to seed node SEED
+    And I have mining node MINER connected to base node SEED and wallet WALLET1
+    And I have a base node SYNCER connected to all seed nodes
+    And I have mine-before-tip mining node MINER2 connected to base node SYNCER and wallet WALLET2
+    And I stop SYNCER
+    When I mine 1 blocks on SEED
+    Then node SEED is at height 1
+    When Mining node MINER mines 99 blocks with min difficulty 2 and max difficulty 100000
+    Then node SEED is at height 100
+    When I start SYNCER
+    And Mining node MINER2 mines 5 blocks with min difficulty 1 and max difficulty 100000
```

Comment:
**NIT:**
Proposal to make it easy to verify in the logs if the miner adhered to these limits
```
... with min difficulty 1 and max difficulty 5
```